modules/ROOT/pages/patterns/variable-length-patterns.adoc

```diff
@@ -0,0 +1,544 @@
+Here is what those segments look like when concatenated with the first repetition:
+
+[source, role=noheader]
+----
+(:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(:Stop)
+(:Stop)-[:NEXT]->(:Stop)
+(:Stop)-[:CALLS_AT]->(:Station { name: 'Clapham Junction' })
+----
+
+The original `MATCH` clause now has the following three parts:
+
+image::patterns_qpp_query_breakdown.svg[]
+
+Translating the union of fixed-length path patterns into a quantified path pattern results in a pattern that will return the correct paths.
+The following query adds a `RETURN` clause that yields the departure and arrival times of the two services:
+
+.Query
+[source, cypher]
+----
+MATCH (:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(d:Stop)
+      ((:Stop)-[:NEXT]->(:Stop)){1,3}
+      (a:Stop)-[:CALLS_AT]->(:Station { name: 'Clapham Junction' })
+RETURN d.departs AS departureTime, a.arrives AS arrivalTime
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+
+| departureTime | arrivalTime
+
+| "17:10Z" | "17:17Z"
+| "17:07Z" | "17:19Z"
+
+2+d|Rows: 2
+
+|===
+
+
+[[quantified-relationships]]
+== Quantified relationships
+_This feature was introduced in Neo4j 5.9._
+
+Quantified relationships allow some simple quantified path patterns to be re-written in a more succinct way.
+Continuing with the example of `Stations` and `Stops` from the previous section, consider the following query:
+
+.Query
+[source,cypher]
+----
+MATCH (d:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(n:Stop)
+      ((:Stop)-[:NEXT]->(:Stop)){1,10}
+      (m:Stop)-[:CALLS_AT]->(a:Station { name: 'Clapham Junction' })
+WHERE m.arrives < time('17:18')
+RETURN n.departs AS departureTime
+----
+
+If the relationship `NEXT` only connects `Stop` nodes, the `:Stop` label expressions can be removed:
+
+.Query
+[source, cypher]
+----
+MATCH (d:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(n:Stop)
+      (()-[:NEXT]->()){1,10}
+      (m:Stop)-[:CALLS_AT]->(a:Station { name: 'Clapham Junction' })
+WHERE m.arrives < time('17:18')
+RETURN n.departs AS departureTime
+----
+
+When the quantified path pattern has one relationship pattern, it can be abbreviated to a _quantified relationship_.
+A quantified relationship is a relationship pattern with a postfix quantifier.
+Below is the previous query rewritten with a quantified relationship:
+
+.Query
+[source, cypher]
+----
+MATCH (d:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-
+        (n:Stop)-[:NEXT]->{1,10}(m:Stop)-[:CALLS_AT]->
+        (a:Station { name: 'Clapham Junction' })
+WHERE m.arrives < time('17:18')
+RETURN n.departs AS departureTime
+----
+
+The scope of the quantifier `{1,10}` is the relationship pattern `-[:NEXT]\->` and not the node patterns abutting it.
+More generally, where a path pattern contained in a quantified path pattern has the following form:
+
+[source, syntax, role=noheader]
+----
+(() <relationship pattern> ()) <quantifier>
```

Comment:
In mathematic papers, there are often lots of parentheses. They tend to "solve this issue" by writing them in different sizes (and sometimes colours). Would that be an option in the documentation?

In this case, the delimiting parentheses of the QPP would be set in a larger font.

Reply:
For mathematics, unless the content overflows, it is a matter of style. See pg 104 of the [AMS style guide](https://www.ams.org/publications/authors/AMS-StyleGuide-online.pdf). In any case, would it not be misleading to suggest anything other than "(" and ")" can be used?